allennlp/data/vocabulary.py

```diff
@@ -9,26 +9,37 @@
 
 class _NamespaceDependentDefaultDict(defaultdict):
     """
-    Using ``defaultdicts`` to handle namespaces is really convenient - you don't need to worry
-    about whether you've seen a particular namespace before in any of the methods in
-    ``Vocabulary``.  But because some namespaces need padding (like "tokens") and some don't (like
-    "labels"), we want different defaults depending on the namespace.  This lets us still use a
-    ``defaultdict``, but have different default values depending on the key that we're given.
+    Sometimes certain namespaces need padding (like "tokens") and some don't (like
+    "labels"), and we want different defaults depending on the namespace.  This class lets us use a
+    ``defaultdict`` (https://docs.python.org/2/library/collections.html#collections.defaultdict),
```

Comment:
Actually, I think we can just do this whole thing a little simpler, by constructing a lambda function in `Vocabulary.__init__`.  With your namespace matching function, this lambda function is now simple enough to be practical.  I'll do that in a separate PR once this one is merged - so don't worry too much about fixing stuff in this particular class.